projects/batfish-common-protocol/src/main/java/org/batfish/storage/FileStorageImpl.java

```diff
@@ -0,0 +1,158 @@
+package org.batfish.storage;
+
+import com.google.common.collect.Sets;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.batfish.common.BatfishException;
+import org.batfish.common.BfConsts;
+import org.batfish.common.util.CommonUtil;
+import org.batfish.datamodel.pojo.Analysis;
+
+public class FileStorageImpl implements Storage {
+
+  private final Path _containersLocation;
+
+  public FileStorageImpl(Path containersLocation) throws BatfishException {
+    try {
+      if (containersLocation != null) {
+        _containersLocation = containersLocation;
+        _containersLocation.toFile().mkdir();
```

Comment:
Do we need to check the existence of all parent directories? Or use mkdirs() function?

Reply:
Done